File: backend/app/models/live_log.py
from sqlalchemy import Column, String, DateTime, ForeignKey, Text
from datetime import datetime, timezone
from app.core.database import Base
import uuid

class LiveLog(Base):
    __tablename__ = "live_logs"

    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()), index=True)
    agent_id = Column(String, ForeignKey("agents.id", ondelete="CASCADE"), nullable=False)
    timestamp = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))
    source_ip = Column(String, index=True, nullable=True)
    log_message = Column(Text, nullable=False)
    threat_level = Column(String, nullable=True) # e.g., INFO, WARN, CRITICAL

File: core/app/Models/LiveLog.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Concerns\HasUuids;

class LiveLog extends Model
{
    use HasFactory, HasUuids;

    protected $fillable = [
        'agent_id',
        'timestamp',
        'source_ip',
        'log_message',
        'threat_level'
    ];

    protected $casts = [
        'timestamp' => 'datetime',
    ];

    public function agent()
    {
        return $this->belongsTo(Agent::class);
    }
}
